### ISSUE TITLE
[rolling-upgrade][centos9] after upgrade `--io-properties-file` isn't set

### ISSUE THREAD

**Issue by fruch** [scylladb/scylladb#19777]
## Packages

Base Scylla version: `6.0.1-20240612.bc89aac9d017` with build-id `e845b7c6530ae1385e7bb3b7def09b4ffdf191c2`
Target Scylla version (or git commit hash): `6.1.0~dev-20240716.dde209390fbd` with build-id `6352c3fdba555c50b97c1510b7ed3d6d89cd63ef`

Kernel Version: `5.14.0-474.el9.x86_64`

## Issue description

- [ ] This issue is a regression.
- [x] It is unknown if this issue is a regression.

since we introduced upgrade tests for centos9,
it's failing cause of:
```
2024-07-14T21:33:32.206+00:00 rolling-upgrade--centos-9-db-node-7f2223ed-0-2      !ERR | scylla[96401]:  [shard  0:main] init - I/O Scheduler is not properly configured! This is a non-supported setup, and performance is expected to be unpredictably bad.
 Reason found: none of --io-properties and --io-properties-file are set.
To properly configure the I/O Scheduler, run the scylla_io_setup utility shipped with Scylla.
To ignore this, see the developer-mode configuration option.
```

for some reason the this options are missing from commandline, for no clear reason.
seem like `scylla_prepare` script isn't get called

## Impact

user start scylla after upgrade, cause some command line argument are missing

## How frequently does it reproduce?

100% of the time on centos-9

## Installation details

Cluster size: 4 nodes (n2-highmem-16)

Scylla Nodes used in this run:
- rolling-upgrade--centos-9-db-node-6a742201-0-4 (34.74.167.86 | 10.142.0.76) (shards: 14)
- rolling-upgrade--centos-9-db-node-6a742201-0-3 (34.75.121.200 | 10.142.0.156) (shards: 14)
- rolling-upgrade--centos-9-db-node-6a742201-0-2 (34.148.96.108 | 10.142.0.155) (shards: 14)
- rolling-upgrade--centos-9-db-node-6a742201-0-1 (34.138.247.172 | 10.142.0.83) (shards: 14)

OS / Image: `https://www.googleapis.com/compute/v1/projects/centos-cloud/global/images/family/centos-stream-9` (gce: undefined_region)

Test: `rolling-upgrade-centos9-test`
Test id: `6a742201-de66-4d9d-b0ce-427ab0ee4b0a`
Test name: `scylla-master/rolling-upgrade/rolling-upgrade-centos9-test`
Test config file(s):

- [rolling-upgrade.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/a5565bc894ca4a534364b7faf02931b79ee5378e/test-cases/upgrades/rolling-upgrade.yaml)

<details>
<summary>
Logs and commands
</summary>

- Restore Monitor Stack command: `$ hydra investigate show-monitor 6a742201-de66-4d9d-b0ce-427ab0ee4b0a`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=6a742201-de66-4d9d-b0ce-427ab0ee4b0a)
- Show all stored logs command: `$ hydra investigate show-logs 6a742201-de66-4d9d-b0ce-427ab0ee4b0a`

## Logs:
- **db-cluster-6a742201.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/6a742201-de66-4d9d-b0ce-427ab0ee4b0a/20240717_011052/db-cluster-6a742201.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/6a742201-de66-4d9d-b0ce-427ab0ee4b0a/20240717_011052/db-cluster-6a742201.tar.gz)
- **sct-runner-events-6a742201.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/6a742201-de66-4d9d-b0ce-427ab0ee4b0a/20240717_011052/sct-runner-events-6a742201.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/6a742201-de66-4d9d-b0ce-427ab0ee4b0a/20240717_011052/sct-runner-events-6a742201.tar.gz)
- **sct-6a742201.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/6a742201-de66-4d9d-b0ce-427ab0ee4b0a/20240717_011052/sct-6a742201.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/6a742201-de66-4d9d-b0ce-427ab0ee4b0a/20240717_011052/sct-6a742201.log.tar.gz)
- **loader-set-6a742201.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/6a742201-de66-4d9d-b0ce-427ab0ee4b0a/20240717_011052/loader-set-6a742201.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/6a742201-de66-4d9d-b0ce-427ab0ee4b0a/20240717_011052/loader-set-6a742201.tar.gz)
- **monitor-set-6a742201.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/6a742201-de66-4d9d-b0ce-427ab0ee4b0a/20240717_011052/monitor-set-6a742201.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/6a742201-de66-4d9d-b0ce-427ab0ee4b0a/20240717_011052/monitor-set-6a742201.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/rolling-upgrade/job/rolling-upgrade-centos9-test/5/)
[Argus](https://argus.scylladb.com/test/09375f78-2fec-44a2-91b2-83ddb2403f14/runs?additionalRuns[]=6a742201-de66-4d9d-b0ce-427ab0ee4b0a)
</details>

**Comment by enaydanov** [comment]
## Packages

Base Scylla version: `6.0.2-20240703.c9cd171f426e` with build-id `f42d84b1312f866e5b14a8a8ecc8ffff0e4d5d19`
Target Scylla version (or git commit hash): `6.1.0-20240804.abbf0b24a60c` with build-id `374d38ae5c8f543c472d3920542bbb543be25d09`

Kernel Version: `5.14.0-474.el9.x86_64`

## Installation details

Cluster size: 4 nodes (n2-highmem-16)

Scylla Nodes used in this run:
- rolling-upgrade--centos-9-db-node-50bbc0d3-0-4 (35.196.199.101 | 10.142.0.169) (shards: 14)
- rolling-upgrade--centos-9-db-node-50bbc0d3-0-3 (34.74.194.33 | 10.142.0.166) (shards: 14)
- rolling-upgrade--centos-9-db-node-50bbc0d3-0-2 (34.148.113.197 | 10.142.0.163) (shards: 14)
- rolling-upgrade--centos-9-db-node-50bbc0d3-0-1 (34.23.84.102 | 10.142.0.162) (shards: 14)


OS / Image: `https://www.googleapis.com/compute/v1/projects/centos-cloud/global/images/family/centos-stream-9` (gce: undefined_region)

Test: `rolling-upgrade-centos9-test`
Test id: `50bbc0d3-6f52-48ff-a733-0b5d05969aa9`
Test name: `scylla-6.1/rolling-upgrade/rolling-upgrade-centos9-test`
Test method: `upgrade_test.UpgradeTest.test_rolling_upgrade`
Test config file(s):

- [rolling-upgrade.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/b874205d8c2a52b37321366cee02d041dc049ca9/test-cases/upgrades/rolling-upgrade.yaml)


<details>
<summary>
Logs and commands
</summary>


- Restore Monitor Stack command: `$ hydra investigate show-monitor 50bbc0d3-6f52-48ff-a733-0b5d05969aa9`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=50bbc0d3-6f52-48ff-a733-0b5d05969aa9)
- Show all stored logs command: `$ hydra investigate show-logs 50bbc0d3-6f52-48ff-a733-0b5d05969aa9`


## Logs:
- **db-cluster-50bbc0d3.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/50bbc0d3-6f52-48ff-a733-0b5d05969aa9/20240806_150645/db-cluster-50bbc0d3.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/50bbc0d3-6f52-48ff-a733-0b5d05969aa9/20240806_150645/db-cluster-50bbc0d3.tar.gz)
- **sct-runner-events-50bbc0d3.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/50bbc0d3-6f52-48ff-a733-0b5d05969aa9/20240806_150645/sct-runner-events-50bbc0d3.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/50bbc0d3-6f52-48ff-a733-0b5d05969aa9/20240806_150645/sct-runner-events-50bbc0d3.tar.gz)
- **sct-50bbc0d3.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/50bbc0d3-6f52-48ff-a733-0b5d05969aa9/20240806_150645/sct-50bbc0d3.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/50bbc0d3-6f52-48ff-a733-0b5d05969aa9/20240806_150645/sct-50bbc0d3.log.tar.gz)
- **loader-set-50bbc0d3.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/50bbc0d3-6f52-48ff-a733-0b5d05969aa9/20240806_150645/loader-set-50bbc0d3.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/50bbc0d3-6f52-48ff-a733-0b5d05969aa9/20240806_150645/loader-set-50bbc0d3.tar.gz)
- **monitor-set-50bbc0d3.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/50bbc0d3-6f52-48ff-a733-0b5d05969aa9/20240806_150645/monitor-set-50bbc0d3.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/50bbc0d3-6f52-48ff-a733-0b5d05969aa9/20240806_150645/monitor-set-50bbc0d3.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-6.1/job/rolling-upgrade/job/rolling-upgrade-centos9-test/6/)
[Argus](https://argus.scylladb.com/test/fd1b18c2-f8b4-4ad3-9479-0001715eaff2/runs?additionalRuns[]=50bbc0d3-6f52-48ff-a733-0b5d05969aa9)
</details>

**Comment by enaydanov** [comment]
@mykaul should we set 6.1 milestone to this issue?

**Comment by mykaul** [comment]
@syuu1228 - did you have a chance to look at this?

**Comment by roydahan** [comment]
@yaronkaikov , @syuu1228 this is urgent for 6.1, please take a look.

**Comment by syuu1228** [comment]
I tried to reproduce this both on local VM and GCE, but could not reproduce it.
Maybe it is because difference of executed command so I'm still checking Jenkins logs.

Also, tried to run same artifact test on Jenkins, it doesn't cause same error but got different error:
https://jenkins.scylladb.com/view/master/job/scylla-master/job/releng-testing/job/rolling-upgrade/job/rolling-upgrade-centos9-test/5/

The error message "I/O Scheduler is not properly configured!" is looks like /etc/scylla.d/io.conf is deleted or reverted to default setting (no parameter).
But when we upgrade by yum it won't automatically overwrite since we specify such parameter on .spec file, and we recently haven't changed these settings, so it unlikely rpm packaging issue.

**Comment by syuu1228** [comment]
Ah, I finally understood how it occur - it is not happen when upgrade test, it is actually happen in downgrade test after upgrade test.

On downgrade test, it run `sudo yum remove scylla-\*cqlsh -y`, `sudo yum downgrade scylla-\*cqlsh -y`, and `sudo yum install scylla -y` to downgrade the package.

```
< t:2024-07-17 01:07:59,972 f:base.py         l:143  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Command "sudo yum downgrade scylla-\*cqlsh -y" finished with status 0
< t:2024-07-17 01:07:59,972 f:remote_base.py  l:521  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Running command "sudo yum remove scylla-\*cqlsh -y"...
< t:2024-07-17 01:08:00,085 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.28>: com.datastax.driver.core.exceptions.TransportException: [rolling-upgrade--centos-9-db-node-6a742201-0-4.c.sct-project-1.internal/10.142.0.76:9042] Connection has been closed
< t:2024-07-17 01:08:00,291 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Dependencies resolved.
< t:2024-07-17 01:08:00,292 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: ================================================================================
< t:2024-07-17 01:08:00,292 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  Package              Arch   Version                              Repo     Size
< t:2024-07-17 01:08:00,292 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: ================================================================================
< t:2024-07-17 01:08:00,292 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Removing:
< t:2024-07-17 01:08:00,292 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla-cqlsh         x86_64 6.1.0~dev-0.20240716.dde209390fbd    @scylla 6.7 M
< t:2024-07-17 01:08:00,293 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Removing dependent packages:
< t:2024-07-17 01:08:00,293 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla               x86_64 6.1.0~dev-0.20240716.dde209390fbd    @scylla   0--
< t:2024-07-17 01:08:00,293 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Removing unused dependencies:
< t:2024-07-17 01:08:00,293 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla-conf          x86_64 6.1.0~dev-0.20240716.dde209390fbd    @scylla  28 k
< t:2024-07-17 01:08:00,293 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla-kernel-conf   x86_64 6.1.0~dev-0.20240716.dde209390fbd    @scylla 5.1 k
< t:2024-07-17 01:08:00,293 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla-node-exporter x86_64 6.1.0~dev-0.20240716.dde209390fbd    @scylla  13 M
< t:2024-07-17 01:08:00,293 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla-python3       x86_64 6.1.0~dev-0.20240716.dde209390fbd    @scylla  77 M
< t:2024-07-17 01:08:00,293 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla-server        x86_64 6.1.0~dev-0.20240716.dde209390fbd    @scylla 182 M
< t:2024-07-17 01:08:00,293 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:-
< t:2024-07-17 01:08:00,293 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: 

Transaction Summary
< t:2024-07-17 01:08:00,293 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: ================================================================================
< t:2024-07-17 01:08:00,293 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Remove  7 Packages

...


< t:2024-07-17 01:08:02,359 f:remote_base.py  l:521  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Running command "sudo yum downgrade scylla\* -y"...
< t:2024-07-17 01:08:02,367 f:db_log_reader.py l:125  c:sdcm.db_log_reader   p:DEBUG > 2024-07-17T01:08:02.290+00:00 rolling-upgrade--centos-9-db-node-6a742201-0-4     !INFO | sudo[105693]: pam_unix(sudo:session): session closed for user root
< t:2024-07-17 01:08:02,587 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.28>: com.datastax.driver.core.exceptions.TransportException: [rolling-upgrade--centos-9-db-node-6a742201-0-4.c.sct-project-1.internal/10.142.0.76:9042] Connection has been closed
< t:2024-07-17 01:08:02,869 f:db_log_reader.py l:125  c:sdcm.db_log_reader   p:DEBUG > 2024-07-17T01:08:02.790+00:00 rolling-upgrade--centos-9-db-node-6a742201-0-4   !NOTICE | sudo[105818]: scylla-test : PWD=/home/scylla-test ; USER=root ; COMMAND=/bin/yum downgrade scylla* -y
< t:2024-07-17 01:08:02,870 f:db_log_reader.py l:125  c:sdcm.db_log_reader   p:DEBUG > 2024-07-17T01:08:02.790+00:00 rolling-upgrade--centos-9-db-node-6a742201-0-4     !INFO | sudo[105818]: pam_unix(sudo:session): session opened for user root(uid=0) by scylla-test(uid=1000)
< t:2024-07-17 01:08:02,920 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Last metadata expiration check: 0:00:06 ago on Wed 17 Jul 2024 01:07:56 AM UTC.
< t:2024-07-17 01:08:03,024 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.28>: total,       1192490,   13378,   13378,   13378,    74.1,    56.4,   199.5,   388.5,   552.1,   622.3,  110.0,  0.05034,      0,      0,       0,       0,       0,       0
< t:2024-07-17 01:08:03,025 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.28>: com.datastax.driver.core.exceptions.TransportException: [rolling-upgrade--centos-9-db-node-6a742201-0-4.c.sct-project-1.internal/10.142.0.76:9042] Connection has been closed
< t:2024-07-17 01:08:03,041 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Dependencies resolved.
< t:2024-07-17 01:08:03,042 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: ================================================================================
< t:2024-07-17 01:08:03,042 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  Package           Arch    Version                          Repository     Size
< t:2024-07-17 01:08:03,042 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: ================================================================================
< t:2024-07-17 01:08:03,042 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Downgrading:
< t:2024-07-17 01:08:03,042 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla-debuginfo  x86_64  6.0.1-0.20240612.bc89aac9d017    scylladb-6.0  311 M
< t:2024-07-17 01:08:03,042 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:-
< t:2024-07-17 01:08:03,042 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Transaction Summary
< t:2024-07-17 01:08:03,042 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: ================================================================================
< t:2024-07-17 01:08:03,042 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Downgrade  1 Package

...


< t:2024-07-17 01:09:08,569 f:remote_base.py  l:521  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Running command "sudo yum install scylla -y"...
< t:2024-07-17 01:09:08,609 f:db_log_reader.py l:125  c:sdcm.db_log_reader   p:DEBUG > 2024-07-17T01:09:08.540+00:00 rolling-upgrade--centos-9-db-node-6a742201-0-4     !INFO | sudo[105931]: pam_unix(sudo:session): session closed for user root
< t:2024-07-17 01:09:09,065 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.28>: com.datastax.driver.core.exceptions.TransportException: [rolling-upgrade--centos-9-db-node-6a742201-0-4.c.sct-project-1.internal/10.142.0.76:9042] Connection has been closed
< t:2024-07-17 01:09:09,111 f:db_log_reader.py l:125  c:sdcm.db_log_reader   p:DEBUG > 2024-07-17T01:09:09.041+00:00 rolling-upgrade--centos-9-db-node-6a742201-0-4   !NOTICE | sudo[105935]: scylla-test : PWD=/home/scylla-test ; USER=root ; COMMAND=/bin/yum install scylla -y
< t:2024-07-17 01:09:09,111 f:db_log_reader.py l:125  c:sdcm.db_log_reader   p:DEBUG > 2024-07-17T01:09:09.041+00:00 rolling-upgrade--centos-9-db-node-6a742201-0-4     !INFO | sudo[105935]: pam_unix(sudo:session): session opened for user root(uid=0) by scylla-test(uid=1000)
< t:2024-07-17 01:09:09,125 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Last metadata expiration check: 0:01:13 ago on Wed 17 Jul 2024 01:07:56 AM UTC.
< t:2024-07-17 01:09:09,245 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Dependencies resolved.
< t:2024-07-17 01:09:09,246 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: ========================================================================================
< t:2024-07-17 01:09:09,246 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  Package                Arch    Version                       Repository            Size
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: ========================================================================================
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Installing:
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla                 x86_64  6.0.1-0.20240612.bc89aac9d017 scylladb-6.0         6.9 k
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Installing dependencies:
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla-conf            x86_64  6.0.1-0.20240612.bc89aac9d017 scylladb-6.0          18 k
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla-cqlsh           noarch  6.0.1-0.20240612.bc89aac9d017 scylladb-6.0-noarch  1.3 M
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla-kernel-conf     x86_64  6.0.1-0.20240612.bc89aac9d017 scylladb-6.0          12 k
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla-node-exporter   x86_64  6.0.1-0.20240612.bc89aac9d017 scylladb-6.0         4.4 M
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla-python3         x86_64  6.0.1-0.20240612.bc89aac9d017 scylladb-6.0          21 M
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:  scylla-server          x86_64  6.0.1-0.20240612.bc89aac9d017 scylladb-6.0          55 M
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>:-
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Transaction Summary
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: ========================================================================================
< t:2024-07-17 01:09:09,247 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Install  7 Packages

```

At that time /etc/scylla.d/io.conf is removed, we need to restore configuration backup.
It does in following command:
```
< t:2024-07-17 01:09:24,437 f:remote_base.py  l:521  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Running command "for conf in $( rpm -qc $(rpm -qa | grep scylla) | grep -v contains ) /etc/systemd/system/{var-lib-scylla,var-lib-systemd-coredump}.mount; do if test -e $conf.backup; then sudo cp -v $conf.backup $conf; fi; done"...
```

However, backup is actually created different name ($conf.autobackup, not $conf.backup):
```
< t:2024-07-17 01:09:24,437 f:remote_base.py  l:521  c:RemoteLibSSH2CmdRunner p:DEBUG > <10.142.0.76>: Running command "for conf in $( rpm -qc $(rpm -qa | grep scylla) | grep -v contains ) /etc/systemd/system/{var-lib-scylla,var-lib-systemd-coredump}.mount; do if test -e $conf.backup; then sudo cp -v $conf.backup $conf; fi; done"...
```

I checked SCT source code, I found that backup and restore filename are inconsistent:
```
def backup_conf(node):
    if node.distro.is_rhel_like:
        node.remoter.run(
            r'for conf in $( rpm -qc $(rpm -qa | grep scylla) | grep -v contains ) '
            r'/etc/systemd/system/{var-lib-scylla,var-lib-systemd-coredump}.mount; '
            r'do if test -e $conf; then sudo cp -v $conf $conf.autobackup; fi; done')
    else:
        node.remoter.run(
            r'for conf in $(cat /var/lib/dpkg/info/scylla-*server.conffiles '
            r'/var/lib/dpkg/info/scylla-*conf.conffiles '
            r'/var/lib/dpkg/info/scylla-*jmx.conffiles | grep -v init ) '
            r'/etc/systemd/system/{var-lib-scylla,var-lib-systemd-coredump}.mount; '
            r'do if test -e $conf; then sudo cp -v $conf $conf.backup; fi; done')


def recover_conf(node):
    if node.distro.is_rhel_like:
        node.remoter.run(
            r'for conf in $( rpm -qc $(rpm -qa | grep scylla) | grep -v contains ) '
            r'/etc/systemd/system/{var-lib-scylla,var-lib-systemd-coredump}.mount; '
            r'do if test -e $conf.backup; then sudo cp -v $conf.backup $conf; fi; done')
    else:
        node.remoter.run(
            r'for conf in $(cat /var/lib/dpkg/info/scylla-*server.conffiles '
            r'/var/lib/dpkg/info/scylla-*conf.conffiles '
            r'/var/lib/dpkg/info/scylla-*jmx.conffiles | grep -v init ); do '
            r'if test -e $conf.backup; then sudo cp -v $conf.backup $conf; fi; done')
```
https://github.com/scylladb/scylla-cluster-tests/blob/master/upgrade_test.py#L100

And seems like this inconsistent introduced at: 
https://github.com/scylladb/scylla-cluster-tests/commit/e3bdc706c055d6164b5bc716016694446bdd3d17

So it seems like SCT problem.

**Comment by fruch** [comment]
no @yaronkaikov, it's for @syuu1228 to fix.

it starts with the change of removing scylla-jmx, that got @aleksbykov to fix the affect of it.

**Comment by yaronkaikov** [comment]
I don't understand how it's related to @syuu1228, but anyway, I removed the `release-blocker` as it's an issue with SCT

@roydahan please assign  someone to handle this, as i will be on PTO starting tomorrow

**Comment by fruch** [comment]
> I don't understand how it's related to @syuu1228, but anyway, I removed the `release-blocker` as it's an issue with SCT
> 
> @roydahan please assign someone to handle this, as i will be on PTO starting tomorrow

https://github.com/scylladb/scylladb/pull/18487

**Comment by roydahan** [comment]
I commented on @aleksbykov commit, we probably need to revert it only in branch-6.1

**Comment by mykaul** [comment]
@roydahan - what's the next step here?

**Comment by roydahan** [comment]
I can't find the discussion we had on the commit.
@fruch IIRC you wrote that we can just change name from of the 'autobackup'.
Who is doing it?
@syuu1228 / @aleksbykov / @fruch ?

**Comment by aleksbykov** [comment]
@roydahan , i prepared the fix: https://github.com/scylladb/scylla-cluster-tests/pull/8372. But currently, it is blocked to run with tablets + counters. ( counters are not supported with tablets)

**Comment by fruch** [comment]
> @roydahan , i prepared the fix: [scylladb/scylla-cluster-tests#8372](https://github.com/scylladb/scylla-cluster-tests/pull/8372). But currently, it is blocked to run with tablets + counters. ( counters are not supported with tablets)

didn't we attend to disabling counters related thing from upgrade tests already ? whatchanged ?
